3.x: RackAwareRoundRobinPolicyTest.should_only_use_remote_hosts_when_using_non_dc_local_cl can fail

Run: https://github.com/scylladb/java-driver/actions/runs/12054939831/job/33614160482?pr=385

Output:
```
Error:  Tests run: 1277, Failures: 1, Errors: 0, Skipped: 652, Time elapsed: 741.583 s <<< FAILURE! - in TestSuite
Error:  com.datastax.driver.core.policies.RackAwareRoundRobinPolicyTest.should_only_use_remote_hosts_when_using_non_dc_local_cl[ALL](6)  Time elapsed: 0.067 s  <<< FAILURE!
java.lang.AssertionError: Query fail
	at org.testng.Assert.fail(Assert.java:83)
	at com.datastax.driver.core.QueryTracker.query(QueryTracker.java:106)
	at com.datastax.driver.core.QueryTracker.query(QueryTracker.java:72)
	at com.datastax.driver.core.policies.RackAwareRoundRobinPolicyTest.should_only_use_remote_hosts_when_using_non_dc_local_cl(RackAwareRoundRobinPolicyTest.java:625)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:200)
	at org.jboss.byteman.contrib.bmunit.BMNGAbstractRunner.run(BMNGAbstractRunner.java:56)
	at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:707)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:151)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:192)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:93)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.util.concurrent.ExecutionException: com.datastax.driver.core.exceptions.TransportException: [/127.0.1.1:43375] Connection has been closed
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:476)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:380)
	at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:180)
	at com.datastax.driver.core.QueryTracker.query(QueryTracker.java:94)
	... 37 more
Caused by: com.datastax.driver.core.exceptions.TransportException: [/127.0.1.1:43375] Connection has been closed
	at com.datastax.driver.core.Connection$ConnectionCloseFuture.force(Connection.java:1746)
	at com.datastax.driver.core.Connection$ConnectionCloseFuture.force(Connection.java:1727)
	at com.datastax.driver.core.CloseFuture$Forwarding.force(CloseFuture.java:88)
	at com.datastax.driver.core.SessionManager.onDown(SessionManager.java:534)
	at com.datastax.driver.core.Cluster$Manager.onDown(Cluster.java:2244)
	at com.datastax.driver.core.Cluster$Manager.access$1200(Cluster.java:1597)
	at com.datastax.driver.core.Cluster$Manager$5.runMayThrow(Cluster.java:2199)
	at com.datastax.driver.core.ExceptionCatchingRunnable.run(ExceptionCatchingRunnable.java:32)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:108)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:41)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:750)
```

--- a/driver-core/src/test/java/com/datastax/driver/core/ScassandraCluster.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/ScassandraCluster.java
@@ -16,6 +16,7 @@
 package com.datastax.driver.core;
 
 import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.ConditionChecker.check;
 import static org.scassandra.cql.MapType.map;
 import static org.scassandra.cql.PrimitiveType.BIG_INT;
 import static org.scassandra.cql.PrimitiveType.BOOLEAN;
@@ -45,6 +46,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
+import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 import org.scassandra.Scassandra;
 import org.scassandra.ScassandraFactory;
@@ -328,12 +330,32 @@ public class ScassandraCluster {
   public void stop(Cluster cluster, int node) {
     logger.debug("Stopping node {}.", node);
     Scassandra scassandra = node(node);
+    Host host = TestUtils.findHost(cluster, node);
     try {
       scassandra.stop();
     } catch (Exception e) {
       logger.error("Could not stop node " + scassandra, e);
     }
     assertThat(cluster).host(node).goesDownWithin(60, TimeUnit.SECONDS);
+    waitUntilPoolsAreRemoved(cluster, host);
+  }
+
+  private void waitUntilPoolsAreRemoved(final Cluster cluster, final Host host) {
+    if (host == null) return;
+
+    check()
+        .before(60, TimeUnit.SECONDS)
+        .that(
+            new Callable<Boolean>() {
+              @Override
+              public Boolean call() {
+                for (SessionManager session : cluster.manager.sessions) {
+                  if (session.pools.containsKey(host)) return false;
+                }
+                return true;
+              }
+            })
+        .becomesTrue();
   }
 
   /**

